### PR TITLE
Fix staging/src/k8s.io/apiserver/pkg/server/filters flake test timeout

### PR DESCRIPTION
**Pull request by wzshiming**
#### What type of PR is this?

/kind flake

#### Which issue(s) this PR fixes:

xref https://github.com/kubernetes/kubernetes/issues/98486#issuecomment-783232922

https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/99236/pull-kubernetes-bazel-test/1364043067924942848

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: staging/src/k8s.io/apiserver/pkg/server/filters/priority-and-fairness_test.go [hunk 1]
```diff
@@ -493,6 +507,16 @@ func TestPriorityAndFairnessWithPanicRecoveryAndTimeoutFilter(t *testing.T) {
...
 		rquestTimesOutPath := "/request/time-out-as-designed"
+
+		go func() {
+			select {
+			case <-stopCh:
+			case <-time.After(caseTimeout):
+				close(reqHandlerCompletedCh)
+				t.Error("time out")
+			}
+		}()
```
Comment (tkashem) [inline staging/src/k8s.io/apiserver/pkg/server/filters/priority-and-fairness_test.go:518]: I see this as a hard stopper, the test must complete within a certain timeout. Can we move this to a function block?
```
func hardStop(t *testing.T, stopCh <-chan struct{}, testTimeout time.Duration, cleanup func()) {
	select {
	case <-stopCh:
		// The test has completed normally.
	case <-time.After(testTimeout):
		cleanup()
		t.Fatalf("the test did not finish within %s", testTimeout)
	}
}
```

we can use it 
```
		go hardStop(t, stopCh, requestTimeout*3, func() {
			close(reqHandlerCompletedCh)
		})
```

**Comment by wzshiming** [comment]
@tkashem Thanks. updated.

**Comment by tkashem** [comment]
/lgtm

Thanks!

**Pull request merged**